# MoveIt-specific simulation settings
moveit_sim_hw_interface:
    joint_model_group: controllers_initial_group_
    joint_model_group_pose: controllers_initial_pose_
# Settings for ros_control control loop
generic_hw_control_loop:
    loop_hz: 300
    cycle_time_error_threshold: 0.01
# Settings for ros_control hardware interface
hardware_interface:
    joints:
        - Chasis_Elevador
        - Elevador_Brazo_1
        - Elevador_Brazo_2
        - Brazo_2_Pinza
        - BASE_PINZA_joint
        - Pinza_Garra_D
        - Pinza_Garra_I
        - neck_joint
        - R1
        - R2
        - R3
        - R4
    sim_control_mode: 1  # 0: position, 1: velocity
# Publish all joint states -----------------------------------
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50
# Position Controllers ---------------------------------------
arm_torso_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - Chasis_Elevador
    - Elevador_Brazo_1
    - Elevador_Brazo_2
    - Brazo_2_Pinza
    - BASE_PINZA_joint
  constraints:
    stopped_velocity_tolerance: 0
  gains:
    Chasis_Elevador:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    Elevador_Brazo_1:
      p: 10000
      d: 250
      i: 10
      i_clamp: 1
    Elevador_Brazo_2:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    Brazo_2_Pinza:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    BASE_PINZA_joint:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
torso_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - Elevador_Brazo_1
  constraints:
    stopped_velocity_tolerance: 0
  gains:
    Elevador_Brazo_1:
      p: 10000
      d: 250
      i: 10
      i_clamp: 1
gripper_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - Pinza_Garra_D
    - Pinza_Garra_I
  constraints:
    stopped_velocity_tolerance: 0
  gains:
    Pinza_Garra_D:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    Pinza_Garra_I:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
neck_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - neck_joint
  constraints:
    stopped_velocity_tolerance: 0
  gains:
    neck_joint:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
controller_list:
  - name: arm_torso_controller
    action_ns: follow_joint_trajectory
    type: FollowJointTrajectory
    default: true
    joints:
      - Chasis_Elevador
      - Elevador_Brazo_1
      - Elevador_Brazo_2
      - Brazo_2_Pinza
      - BASE_PINZA_joint
  - name: torso_controller
    action_ns: follow_joint_trajectory
    type: FollowJointTrajectory
    default: true
    joints:
      - Elevador_Brazo_1
  - name: gripper_controller
    action_ns: follow_joint_trajectory
    type: FollowJointTrajectory
    default: true
    joints:
      - Pinza_Garra_D
      - Pinza_Garra_I
  - name: neck_controller
    action_ns: follow_joint_trajectory
    type: FollowJointTrajectory
    default: true
    joints:
      - neck_joint